src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -634,6 +638,109 @@ public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, R
             }
         }
 
+        /// <summary>
+        /// Method to append InstanceAnnotations to the ODataResource and Property
+        /// </summary>
+        /// <param name="resource">param for ODataResource</param>
+        /// <param name="resourceContext">param for ODataResourceContext</param>        
+        public virtual void AppendInstanceAnnotations(ODataResource resource, ResourceContext resourceContext)
+        {
+            IEdmStructuredType structuredType = resourceContext.StructuredType;
+            IEdmStructuredObject structuredObject = resourceContext.EdmObject;
+            PropertyInfo instanceAnnotationInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType,
+                resourceContext.EdmModel);
+
+            object value;
+            
+            if (instanceAnnotationInfo == null || structuredObject == null ||
+                !structuredObject.TryGetPropertyValue(instanceAnnotationInfo.Name, out value) || value == null)
+            {
+                return;
+            }                  
+
+            IDictionary<string, IDictionary<string, object> > instanceAnnotationsDictionary = value as IDictionary<string, IDictionary<string, object>>;
+
+            if(instanceAnnotationsDictionary != null)
+            {
+                if (instanceAnnotationsDictionary.ContainsKey(string.Empty))
+                {
+                    if (resource.InstanceAnnotations == null)
```

Comment:
It's not possible for `InstanceAnnotations` to be null is it?

Reply:
@sreejith - you shouldn't need this check for InstanceAnnotations to be null; it is guaranteed to be non-null by the ODataResource.  You can remove 669-672.

---
In reply to: [474988655](https://github.com/OData/WebApi/pull/2219#discussion_r474988655) [](ancestors = 474988655,473265330)